### PR TITLE
#5822 Fix: deprecated errors

### LINKED ISSUE #5822: Fix 'typescript-eslint/no-deprecated' errors
`typescript-eslint` added new rule `no-deprecated`, it'll be useful for finding and updating all usages of deprecated methods in our code.

It found around 160 deprecation errors - https://flowcrypt.semaphoreci.com/jobs/4d93845e-1561-4254-a759-2ff804ade7c9

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR fixed deprecated errors

close #5822 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/common/inject.ts [hunk 1]
```diff
@@ -129,6 +129,7 @@ export class Injector {
           el.remove();
         })
       )
+      // eslint-disable-next-line @typescript-eslint/no-deprecated
       .hover(
```
Comment (sosnovsky) [inline extension/js/common/inject.ts:133]: can we update `.hover` to `.on('hover', ..)` here?

File: extension/js/common/browser/browser.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export class Browser {
       const ev = document.createEvent('MouseEvents');
 
       // @ts-expect-error - safari only. expected 15 arguments, but works well with 4
+      // eslint-disable-next-line @typescript-eslint/no-deprecated
       ev.initMouseEvent('click', true, true, window);
```
Comment (sosnovsky) [inline extension/js/common/browser/browser.ts:29]: can we replace it with `new MouseEvent` like you did in `GmailElementReplacer`?

File: extension/js/common/inject.ts (hunk 1 above)
Comment (ioanmo226) [reply extension/js/common/inject.ts:133]: Yep!

**Review by sosnovsky — APPROVED** [review]
Well done 👍

**Pull request merged**